GooglePlayInstant/LoadingScreen/LoadingBar.cs

```diff
@@ -0,0 +1,174 @@
+
+            var loadingBarOutlineImage = loadingBarOutlineGameObject.GetComponent<Image>();
+            loadingBarOutlineImage.sprite = AssetDatabase.GetBuiltinExtraResource<Sprite>("UI/Skin/InputFieldBackground.psd");
+
+            loadingBarOutlineImage.type = Image.Type.Sliced;
+            loadingBarOutlineImage.fillCenter = false;
+
+            // Set size of component
+            var loadingBarOutlineRectTransform = loadingBarOutlineGameObject.GetComponent<RectTransform>();
+            var loadingBarRectTransform = loadingBarGameObject.GetComponent<RectTransform>();
+            loadingBarOutlineRectTransform.sizeDelta = loadingBarRectTransform.sizeDelta;
+
+            // Position outline component
+            loadingBarOutlineGameObject.transform.position = loadingBarGameObject.transform.position;
+        }
+
+        private static void SetLoadingBarFill(GameObject loadingBarGameObject)
+        {
+            var loadingBarFillGameObject = new GameObject(LoadingBarFillGameObjectName);
+            loadingBarFillGameObject.transform.SetParent(loadingBarGameObject.transform);
+
+            loadingBarFillGameObject.AddComponent<Image>();
+            var loadingBarFillImage = loadingBarFillGameObject.GetComponent<Image>();
+            loadingBarFillImage.color = Color.green;
+
+            // Set size of component
+            var loadingBarFillRectTransform = loadingBarFillGameObject.GetComponent<RectTransform>();
+
+            var loadingBarRectTransform = loadingBarGameObject.GetComponent<RectTransform>();
+            loadingBarFillRectTransform.sizeDelta = new Vector2(0,
+                loadingBarRectTransform.sizeDelta.y - LoadingBarFillPadding);
+
+            // Position fill component
+            loadingBarFillGameObject.transform.position = loadingBarGameObject.transform.position;
+        }
+#endif
+
+        /// <summary>
+        /// Updates a loading bar by the progress made by an asynchronous operation up to a specific percentage of
+        /// the loading bar. 
+        /// </summary>
+        public static IEnumerator UpdateLoadingBar(AsyncOperation operation, float percentageOfLoadingBar)
+        {
+            var loadingBarRectTransform = GameObject.Find(LoadingBarGameObjectName).GetComponent<RectTransform>();
+
+            var loadingBarFillRectTransform =
+                GameObject.Find(LoadingBarFillGameObjectName).GetComponent<RectTransform>();
+
+            // Total amount of space the loading bar can occupy
+            var loadingBarFillMaxWidth = loadingBarRectTransform.sizeDelta.x - LoadingBarFillPadding;
+
+            // Percentage of space that is allocated for this async operation
+            var loadingMaxWidth = loadingBarFillMaxWidth * percentageOfLoadingBar;
+
+            // Current width of the loading bar
+            var currentLoadingBarFill = loadingBarFillRectTransform.sizeDelta.x;
+
+            var loadingIsDone = false;
```

Comment:
Do you need this variable? I may be missing something, but I think you can remove it and just check operation.isDone

Reply:
The reason operation.isDone wasn't sufficient when I was trying to do the loading bar is because the while loop would exit without updating the loading bar to completion, which was necessary (especially in the case of having two different percentage uses for the loading bar. If one operation didn't update the loading bar to completion, the next operation would be off). 

What I did was remove the operation.isDone from the condition for the while loop -- it was unnecessary. However, I wasn't too sure how to approach the issue of completing the loading bar and completing the operation even after the while loop exited. 

Ex thread for what my issue was: https://stackoverflow.com/questions/29475232/standard-idiom-for-executing-a-while-loop-one-more-time

Reply:
Resolved offline: going through with flag.